packages/delegation-toolkit/src/actions/signUserOperationAction.ts

```diff
@@ -0,0 +1,148 @@
+  /** Account to sign with */
+  account?: Account | Address;
+  /** The user operation to sign */
+  userOperation: Omit<UserOperationV07, 'signature'>;
+  /** The entry point contract address */
+  entryPoint: { address: Address };
+  /** The chain ID that the entry point is deployed on */
+  chainId: number;
+  /** The address of the smart account */
+  address: Address;
+  /** The name of the domain of the implementation contract */
+  name: 'HybridDeleGator' | 'MultiSigDeleGator';
+  /** The version of the domain of the implementation contract */
+  version?: string;
+};
+
+export type SignUserOperationReturnType = `0x${string}`;
+
+/**
+ * Signs a user operation using a wallet client.
+ * @param client - The wallet client to use for signing.
+ * @param parameters - The parameters for signing the user operation.
+ * @returns The signature of the user operation.
+ * @example
+ * ```ts
+ * const walletClient = createWalletClient({
+ *   chain: mainnet,
+ *   transport: http()
+ * }).extend(signUserOperationAction());
+ *
+ * const signature = await walletClient.signUserOperation({
+ *   userOperation: {
+ *     sender: '0x...',
+ *     nonce: 0n,
+ *     callData: '0x',
+ *     callGasLimit: 1000000n,
+ *     verificationGasLimit: 1000000n,
+ *     preVerificationGas: 21000n,
+ *     maxFeePerGas: 1000000000n,
+ *     maxPriorityFeePerGas: 1000000000n
+ *   },
+ *   entryPoint: { address: '0x...' },
+ *   chainId: 1,
+ *   address: '0x...',
+ *   name: 'HybridDeleGator'
+ * });
+ * ```
+ */
+export async function signUserOperation<
+  TChain extends Chain | undefined,
+  TAccount extends Account | undefined,
+>(
+  client: Client<Transport, TChain, TAccount> & {
+    signTypedData: WalletClient['signTypedData'];
+  },
+  parameters: SignUserOperationParameters,
+): Promise<SignUserOperationReturnType> {
+  const {
+    account: accountParam = client.account,
+    userOperation,
+    entryPoint,
+    chainId,
+    name,
+    address,
+    version = '1',
+  } = parameters;
+
+  if (!accountParam) {
+    throw new BaseError('Account not found. Please provide an account.');
+  }
+
+  const account = parseAccount(accountParam);
+
+  const packedUserOp = toPackedUserOperation({
+    ...userOperation,
+    signature: '0x',
+  });
+
+  return client.signTypedData({
+    account,
+    domain: {
+      chainId,
+      name,
+      version,
+      verifyingContract: address,
+    },
+    types: SIGNABLE_USER_OP_TYPED_DATA,
+    primaryType: 'PackedUserOperation',
+    message: { ...packedUserOp, entryPoint: entryPoint.address },
+  });
+}
+
+/**
+ * Creates a sign user operation action that can be used to extend a wallet client.
+ * @returns A function that can be used with wallet client extend method.
+ * @example
+ * ```ts
+ * const walletClient = createWalletClient({
+ *   chain: mainnet,
+ *   transport: http()
+ * }).extend(signUserOperationAction());
+ * ```
+ */
+export function signUserOperationAction() {
```

Comment:
I think the pattern we've used elsewhere for actions is:

export a function named `XXXAction` that accepts the `client, args` and returns the result. Export this function from the SDK _without_ the `Action` suffix.

We only add the additional layer of the anonymous function when we're explicitly extending a client.

Either way, we should be as consistent with Viem's intention as possible (although the internals are less important than the external SDK surface).